frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.unit.spec.tsx

```diff
@@ -0,0 +1,167 @@
+  visibility_type: "hidden",
+});
+const REVIEWS_TABLE = createReviewsTable();
+
+interface SetupOpts {
+  hideOrdersTable?: boolean;
+}
+
+function setup({ hideOrdersTable = false }: SetupOpts = {}) {
+  setupDatabasesEndpoints([]);
+  setupActionsEndpoints([]);
+  const productsId = findField(PRODUCTS_TABLE.fields, "ID");
+  const ordersProductId = findField(ORDERS_TABLE.fields, "PRODUCT_ID");
+  const reviewsProductId = findField(REVIEWS_TABLE.fields, "PRODUCT_ID");
+  setupTableEndpoints(PRODUCTS_TABLE, [
+    {
+      origin: ordersProductId,
+      origin_id: ordersProductId.id as number,
+      destination: productsId,
+      destination_id: productsId.id as number,
+      relationship: "Mt1",
+    },
+    {
+      origin: reviewsProductId,
+      origin_id: reviewsProductId.id as number,
+      destination: productsId,
+      destination_id: productsId.id as number,
+      relationship: "Mt1",
+    },
+  ]);
+  setupForeignKeyCountQueryEndpoints();
+
+  const ROW_ID_INDEX = 0;
+  const state = createMockState({
+    entities: createMockEntitiesState({
+      tables: [
+        PRODUCTS_TABLE,
+        hideOrdersTable ? HIDDEN_ORDERS_TABLE : ORDERS_TABLE,
+        REVIEWS_TABLE,
+      ],
+    }),
+    qb: createMockQueryBuilderState({
+      card: createMockCard({
+        dataset_query: {
+          database: SAMPLE_DB_ID,
+          type: "query",
+          query: {
+            "source-table": PRODUCTS_ID,
+          },
+        },
+      }),
+      zoomedRowObjectId: testDataset.rows[0][ROW_ID_INDEX] as string,
+      queryResults: [
+        createMockDataset({
+          data: testDataset,
+        }),
+      ],
+    }),
+  });
+  renderWithProviders(
+    <ObjectDetail
+      data={testDataset}
+      settings={{}}
+      isObjectDetail
+      onVisualizationClick={jest.fn()}
+      visualizationIsClickable={jest.fn()}
+    />,
+    {
+      storeInitialState: state,
+    },
+  );
+}
+
+function setupForeignKeyCountQueryEndpoints() {
+  fetchMock.post(
+    {
+      name: "ordersCountQuery",
+      url: "path:/api/dataset",
+      matchPartialBody: true,
+      body: {
+        query: { "source-table": ORDERS_TABLE.id },
+      },
+    },
+    createMockDataset({
+      status: "completed",
+      data: {
+        rows: [[93]],
+      },
+    }),
+  );
+  fetchMock.post(
+    {
+      name: "reviewsCountQuery",
+      url: "path:/api/dataset",
+      matchPartialBody: true,
+      body: {
+        query: { "source-table": REVIEWS_TABLE.id },
+      },
+    },
+    createMockDataset({
+      status: "completed",
+      data: {
+        rows: [[8]],
+      },
+    }),
+  );
+}
+
+function findField(fields: Field[] | undefined, name: string): Field {
+  return fields?.find(field => field.name === name) as Field;
+}
+
+describe("ObjectDetail", () => {
+  it("should render foreign key count when no table is hidden", async () => {
+    setup();
+
+    expect(
+      await screen.findByText(getBrokenUpTextMatcher("8Reviews")),
```

Comment:
nit: This makes it a little bit more readable:

```suggestion
      await screen.findByText(getBrokenUpTextMatcher(["8", "Reviews"].join(""))),
```

Reply:
Good idea.

Reply:
Fixed in https://github.com/metabase/metabase/pull/33697/commits/9b97b91957cdb5324e234c3fb64f1e80ba38b9da